rest-service/manager_rest/resource_manager.py

```diff
@@ -107,6 +108,76 @@ def list_executions(self, include=None, is_include_system_workflows=False,
             get_all_results=get_all_results
         )
 
+    @staticmethod
+    def evaluate_summary_deployment_status(latest_execution_status,
+                                           installation_status):
+        """
+        This method will evaluate the overall deployment status based on
+        different statuses
+        :param latest_execution_status: The latest exectuion status for
+        deployment
+        :param installation_status: The installation status for deployment
+        :return: deployment_status: Overall deployment status
+        """
+        # TODO we need to cover also aggregated statuses for deployment
+        #  children later on
+        if latest_execution_status == DeploymentState.CANCELLED:
+            if installation_status == DeploymentState.ACTIVE:
+                return DeploymentState.GOOD
+            return DeploymentState.REQUIRE_ATTENTION
+        elif latest_execution_status == DeploymentState.IN_PROGRESS and \
+                installation_status == DeploymentState.INACTIVE:
+            return DeploymentState.IN_PROGRESS
```

Comment:
Isn't that handled in `else:`?

Reply:
@mateumann, well the reason for that because I want to check both case `in_progress` && `inactive` together if I remove it and depend only on else I will have unexpected behavior if at least `inactive` is exist it will be be marked as required_attention which should not be, check the next elif which is
```
        elif latest_execution_status == DeploymentState.FAILED or \
                installation_status == DeploymentState.INACTIVE:
            return DeploymentState.REQUIRE_ATTENTION
``` 
removing the above elif check will be always make `inactive to be  `DeploymentState.REQUIRE_ATTENTION`

Reply:
@mabuaisha you're right of course, sorry for the confusion.